src/mbgl/text/collision_feature.cpp

```diff
@@ -7,19 +7,80 @@ namespace mbgl {
 
 CollisionFeature::CollisionFeature(const GeometryCoordinates& line,
                                    const Anchor& anchor,
-                                   const float top,
-                                   const float bottom,
-                                   const float left,
-                                   const float right,
-                                   const std::optional<Padding>& collisionPadding,
+                                   const Shaping& shapedText,
                                    const float boxScale,
                                    const float padding,
                                    const style::SymbolPlacementType placement,
-                                   IndexedSubfeature indexedFeature_,
+                                   const RefIndexedSubfeature& indexedFeature_,
```

Comment:
I'm curious why this changed? "Ref" imposes some lifetime restrictions on index it's using - are those guaranteed to hold? Why TextFit changes this?